src/cmd/usb.js

```diff
@@ -186,52 +187,51 @@ module.exports = class UsbCommand extends CLICommandBase {
 			.then(() => console.log('Done.'));
 	}
 
-	cloudStatus(args, started = Date.now()){
+	async cloudStatus(args) {
 		const { until, timeout, params: { device } } = args;
+		this.newSpin('Querying device...').start();
+		let status = null;
 
-		if (Date.now() - (started + timeout) > 0){
-			throw new Error('timed-out waiting for status...');
+		try {
+			status = await executeWithUsbDevice({
+				args: { idOrName: device }, // device here is the id
+				func: (dev) => this._cloudStatus(dev, until, timeout),
+			});
+		} catch (error) {
+			this.stopSpin();
+			throw new Error(error.message);
+		} finally {
+			this.stopSpin();
+			if (status) {
+				console.log(status.toLowerCase());
+			}
 		}
+	}
 
-		const deviceMgr = {
-			_: null,
-			set(usbDevice){
-				this._ = usbDevice || null;
-				return this;
-			},
-			close(){
-				const { _: usbDevice } = this;
-				return usbDevice ? usbDevice.close() : Promise.resolve();
-			},
-			status(){
-				const { _: usbDevice } = this;
-				let getStatus = usbDevice
-					? usbDevice.getCloudConnectionStatus()
-					: Promise.resolve('unknown');
-
-				return getStatus.then(status => status.toLowerCase());
+	async _cloudStatus(device, until, timeout) {
+		let status = null;
+		const started = Date.now();
+
+		if (until) {
+			while (Date.now() - started < timeout) {
+				try {
+					status = await device.getCloudConnectionStatus();
+					status = status.toLowerCase();
+					if (status === until) {
+						break;
+					}
+				} catch (error) {
+					// ignore error and keep trying until timeout has elapsed
+				}
 			}
-		};
-
-		const queryDevice = openUsbDeviceByIdOrName(device, this._api, this._auth)
-			.then(usbDevice => deviceMgr.set(usbDevice).status());
+			if (Date.now() - started >= timeout) {
+				throw new Error('timed-out waiting for status...');
+			}
+		} else {
+			status = await device.getCloudConnectionStatus();
+		}
 
-		return spin(queryDevice, 'Querying device...')
-			.then(status => {
-				if (until && until !== status){
-					throw new Error(`Unexpected status: ${status}`);
-				}
-				console.log(status);
-			})
-			.catch(error => {
-				if (until){
-					return deviceMgr.close()
-						.then(() => delay(1000))
-						.then(() => this.cloudStatus(args, started));
-				}
-				throw error;
-			})
-			.finally(() => deviceMgr.close());
+		return status;
```

Comment:
How about:

```
const started = Date.now();

if (until) {
	while (Date.now() - started < timeout) {
		try {
			let status = await device.getCloudConnectionStatus();
			status = status.toLowerCase();
			if (status === until) {
				return status;
			}
		} catch (error) {
			// ignore error and keep trying until timeout has elapsed
		}
	}
	throw new Error('timed-out waiting for status...');
} else {
	return device.getCloudConnectionStatus();
}
```

Reply:
this works too. I preferred a single point of return